HighwayPatrol.cpp

```diff
@@ -23,13 +33,36 @@ void HighwayPatrol::scanHighway(Highway* h)
 
 void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
 {
+    std::string flagged;
+
     std::cout << "\n\n";
+
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
+
     if( willArrest )
     {
+        if (dynamic_cast<Car*>(v))
+        {
+            flagged = "car";
+        }
+        if (dynamic_cast<Motorcycle*>(v))
+        {
+            flagged = "motorcycle";
+        }
+        if (dynamic_cast<SemiTruck*>(v))
+        {
+            flagged = "semitruck";
+        }
+        else
+        {
+            flagged = "ILLEGAL VEHICLE";
+        }
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        std::cout << name << ": YOU IN THE [ " << flagged << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
```

Comment:
Figure out why it's printing out `ILLEGAL VEHICLE` instead of one of your vehicle types.
- [ ] revise